datumaro/components/dataset.py

```diff
@@ -905,16 +906,25 @@ def import_from(cls, path: str, format: Optional[str] = None,
         return dataset
 
     @staticmethod
-    def detect(path: str, env: Optional[Environment] = None) -> str:
+    def detect(path: str, env: Optional[Environment] = None,
+            depth: Optional[int] = 2) -> str:
         if env is None:
             env = Environment()
 
-        matches = env.detect_dataset(path)
+        for _ in range(depth):
+            matches = env.detect_dataset(path)
+            if matches and len(matches) == 1:
+                return matches[0]
+
+            paths = glob.glob(osp.join(path, '*'))
+            if len(paths) != 1 or not osp.isdir(paths[0]):
```

Comment:
I suspect, we should also ignore hidden OS files like [`.DS_Store` on Mac](https://perishablepress.com/remove-macosx-ds-store-zip-files-mac/) - I noticed, they often get into the archives.